lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Random.hs

```diff
@@ -322,6 +348,53 @@ hdPassphrase (Key rootXPub) = Passphrase $
     (unXPub rootXPub)
     ("address-hashing" :: ByteString)
 
+toAddress
+    :: Key 'RootK XPub
+    -> Key 'AddressK XPub
+    -> Index 'Hardened 'AccountK
+    -> Index 'Soft 'AddressK
+    -> Address
+toAddress rootXPub (Key addrXPub) accIx addrIx =
+    Address $ encodeBase58 bitcoinAlphabet
+    $ CBOR.toStrictByteString
+    $ encodeAddress addrXPub encodeAddressAttributes
+  where
+    encodeAddressAttributes = mempty
+        <> CBOR.encodeMapLen 1
+        <> CBOR.encodeWord8 1
+        <> encodeDerivationPath rootXPub accIx addrIx
```

Comment:
Note that this is incorrect for `testnet`. The underlying format of an address actually depends on the type of network. See `KeyToAddress` in the AD module.